src/Views/KeyboardLayoutView.vala

```diff
@@ -78,22 +78,34 @@ public class KeyboardLayoutView : AbstractInstallerView {
 
         back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
 
-        next_button.clicked.connect (() => next_step ());
+        next_button.clicked.connect (() => {
+            unowned Gtk.ListBoxRow row = input_variant_widget.main_listbox.get_selected_row ();
+            if (row != null) {
+                var layout = ((LayoutRow) row).layout;
+                Configuration.get_default ().keyboard_layout = layout.name;
```

Comment:
Since you're using `Configuration.get_default ()` multiple times here, I suggest getting the instance once: 
`unowned Configuration = Configuration.get_default ()`.